client/lib/user/user.js

```diff
@@ -107,9 +93,9 @@ User.prototype.initialize = function() {
  * @param {Number} userId The new user ID.
  **/
 User.prototype.clearStoreIfChanged = function( userId ) {
-	const storedUser = store.get( 'wpcom_user' );
+	const storedUserId = store.get( 'wpcom_user_id' );
 
-	if ( storedUser && storedUser.ID !== userId ) {
+	if ( storedUserId != null && storedUserId !== userId ) {
```

Comment:
I suggest making this a bit more explicit; you're relying on behaviour that isn't immediately obvious at first glance. This code is vulnerable to well-intentioned drive-by fixes that want to remove `!=` from the codebase because of lint checks 🙂 

How about:
```suggestion
	if ( storedUserId !== null && storedUserId !== undefined && storedUserId !== userId ) {
```

Reply:
`!= null` is a valid use of the `!=` operator, the [eqeqeq](https://eslint.org/docs/rules/eqeqeq) rule has an `allow-null` exception for it, and our ESLint config has it enabled. I recently learned about it from @ljharb in #36256 🙂